rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -228,6 +236,125 @@ def _decode_curve_deposit_events(
+            raw_sold_amount = hex_or_bytes_to_int(context.tx_log.data[32:64])
+            bought_token_id = hex_or_bytes_to_int(context.tx_log.data[64:96])
+            raw_bought_amount = hex_or_bytes_to_int(context.tx_log.data[96:128])
+            if (
+                # Can only do this check for "normal" exchanges (not exchange_unederlying) since
+                #
+                context.tx_log.topics[0] == TOKEN_EXCHANGE and  # Perform this check only for `TOKEN_EXCHANGE`  # noqa: E501
+                pool_address in self.curve_pools and  # Make sure that the pool is cached
+                len(self.curve_pools[pool_address]) > max(sold_token_id, bought_token_id)  # Make sure that tokens of the pool are cached  # noqa: E501
+            ):
+                sold_token_address = self.curve_pools[pool_address][sold_token_id]
+                bought_token_address = self.curve_pools[pool_address][bought_token_id]
+        else:  # EXCHANGE_MULTIPLE
+            swapping_contract = CURVE_SWAP_ROUTER
+            spender_address = hex_or_bytes_to_address(context.tx_log.topics[1])
+            receiver_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+            raw_sold_amount = hex_or_bytes_to_int(context.tx_log.data[-64:-32])
+            raw_bought_amount = hex_or_bytes_to_int(context.tx_log.data[-32:])
+            # Curve swap router logs route (a list of addresses) that was used. Route consists of
+            # 9 items. The structure is:
+            # 1. Token in
+            # 2..X Pools that were used
+            # X+1. Token out
+            # X+2..9 (unused items) Zero addresses
+            # Here we read only addresses of token in and token out.
+            sold_token_address = hex_or_bytes_to_address(context.tx_log.data[:32])
+            for i in range(1, 9):  # Starting from 1 because at 0 is `sold_token_address`
+                address = hex_or_bytes_to_address(context.tx_log.data[32 * i:32 * (i + 1)])
+                if address == ZERO_ADDRESS:
+                    break
+                bought_token_address = address
+
+        sold_asset: Optional[Asset] = None
+        if sold_token_address is not None:
+            if sold_token_address == ETH_SPECIAL_ADDRESS:
+                sold_asset = A_ETH
+            else:
+                sold_asset = Asset(evm_address_to_identifier(
+                    address=sold_token_address,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ))
+
+        bought_asset: Optional[Asset] = None
+        if bought_token_address is not None:
+            if bought_token_address == ETH_SPECIAL_ADDRESS:
+                bought_asset = A_ETH
+            else:
+                bought_asset = Asset(evm_address_to_identifier(
+                    address=bought_token_address,
+                    chain_id=self.evm_inquirer.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ))
+
```

Comment:
to avoid repeating the code this could be a `_function` in this file. Then is

```
sold_asset = _read_curve_asset(sold_token_address)
bought_asset = _read_curve_asset(bought_token_address)
```